cmake/kokkos_functions.cmake

```diff
@@ -60,6 +60,428 @@ FUNCTION(kokkos_option CAMEL_SUFFIX DEFAULT TYPE DOCSTRING)
 
 ENDFUNCTION()
 
+FUNCTION(kokkos_append_config_line LINE)
+  GLOBAL_APPEND(KOKKOS_TPL_EXPORTS "${LINE}")
+ENDFUNCTION()
+
+MACRO(kokkos_export_cmake_tpl NAME)
+  IF (DEFINED ${NAME}_DIR)
+    KOKKOS_APPEND_CONFIG_LINE("IF(NOT DEFINED ${NAME}_DIR)")
+    KOKKOS_APPEND_CONFIG_LINE("  SET(${NAME}_DIR  ${${NAME}_DIR})")
```

Comment:
This is so confusing.  Please add a comment.

Reply:
Comments added.